lib/rate_map.c

```diff
@@ -235,66 +235,124 @@ rate_map_shift_by_mass(rate_map_t *self, double pos, double mass)
 #error "Base 2 floating point types required"
 #endif
 
+static bool
+valid_sorted_nonempty_array(const double *array, size_t size)
+{
+    size_t idx;
+
+    if ((ptrdiff_t) size < 1 || isnan(array[0])) {
+        return false;
+    }
+    for (idx = 1; idx < size; idx++) {
+        if (!(array[idx - 1] <= array[idx])) {
+            return false;
+        }
+    }
+    return true;
+}
+
+static bool
+fast_search_lookup_valid(fast_search_lookup_t *self)
+{
+    const double *start, *stop;
+
+    if (!(self->query_multiplier >= 0.0)) { // NaN not valid
```

Comment:
Just so I'm clear, you're guarding against ``query_multiplier`` being NaN here by negating >=, right?

Reply:
Yep.